modules/modelSetup/BaseStableDiffusionXLSetup.py

```diff
@@ -214,6 +218,83 @@ def predict(
 
         return model_output_data
 
+    def perceptual_loss(self, generated, target, vgg_model):
```

Comment:
For a full implementation, all these functions should be moved to a central location. That way they can be used by all models, not just SDXL. Maybe modules\util\loss_util.py would be a good place.